nedlibreader/time_test.go

```diff
@@ -0,0 +1,126 @@
+package nedlibreader
+
+import (
+	"testing"
+	"time"
+)
+
+func TestParseValidRFC1123(t *testing.T) {
+	dateString := "Tue, 05 Apr 2024 15:30:00 GMT"
+	expectedTime := time.Date(2024, time.April, 5, 15, 30, 0, 0, time.UTC)
+
+	parsedTime, err := parseTime(dateString)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+		return
+	}
+
+	if !parsedTime.Equal(expectedTime) {
+		t.Errorf("expected time: %v, got: %v", expectedTime, parsedTime)
+	}
+}
+
+func TestParseInvalidRFC1123(t *testing.T) {
+	dateString := "Tue, 05 Apr 2024 15:30:00"
+	expectedError := "failed to parse string as time.Time: Tue, 05 Apr 2024 15:30:00: parsing time \"Tue, 05 Apr 2024 15:30:00\" as \"Mon Jan _2 15:04:05 MST 2006\": cannot parse \", 05 Apr 2024 15:30:00\" as \" \""
```

Comment:
Better to compare error against `time.ParseError` using `errors.As` or `errors.Is`.

Reply:
Fixed in a fixup